src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -39,6 +39,22 @@ impl ScalarFunc {
         Ok(Some(Cow::Owned(format!("{:b}", i).into_bytes())))
     }
 
+    #[inline]
+    pub fn reverse<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string(ctx, row));
```

Comment:
I think here we need to call `eval_string_and_decode` instead of call `str::from_utf8` directly.

Reply:
@hicqu . I've fix this. PTAL.  I made some modifications on `eval_string_and_decode`, disable the `charset` check and use a `TODO` comment instead because different charsets may be pushed down from `TiDB` and some builtins depends on it. `UPPER`, `LOWER` for example.